smithy-build/src/test/java/software/amazon/smithy/build/SmithyBuildTest.java

```diff
@@ -658,4 +656,152 @@ public Node toNode() {
             throw new SourceException("Unable to serialize trait!", SourceLocation.none());
         }
     }
+
+    @Test
+    public void canRunMultiplePluginsWithDifferentArtifactNames() throws URISyntaxException {
+        Map<String, SmithyBuildPlugin> plugins = MapUtils.of("test1", new Test1SerialPlugin(),
+                                                             "test2", new Test2ParallelPlugin());
+        Function<String, Optional<SmithyBuildPlugin>> factory = SmithyBuildPlugin.createServiceFactory();
+        Function<String, Optional<SmithyBuildPlugin>> composed = name -> OptionalUtils.or(
+                Optional.ofNullable(plugins.get(name)), () -> factory.apply(name));
+
+        SmithyBuild builder = new SmithyBuild().pluginFactory(composed);
+        builder.fileManifestFactory(MockManifest::new);
+        builder.config(SmithyBuildConfig.builder()
+                .load(Paths.get(getClass().getResource("artifact-names.json").toURI()))
+                .outputDirectory("/foo")
+                .build());
+
+        SmithyBuildResult results = builder.build();
+        ProjectionResult source = results.getProjectionResult("source").get();
+        ProjectionResult a = results.getProjectionResult("a").get();
+        ProjectionResult b = results.getProjectionResult("b").get();
+
+        assertPluginPresent("test1", "hello1Serial", source, a, b);
+        assertPluginPresent("foo1", "hello1Serial", source, a, b);
+        assertPluginPresent("foo2", "hello1Serial", source, a, b);
+        assertPluginPresent("foo3", "hello2Parallel", source, a, b);
+
+        assertPluginPresent("test2", "hello2Parallel", a, b);
+    }
+
+    @Test
+    public void runsCommand() throws Exception {
+        Assumptions.assumeTrue(isPosix());
+
+        String output = runPosixTestAndGetOutput("run-plugin/run-plugin-test.json", "test-process");
+
+        assertThat(output, containsString("argv1: a"));
+        assertThat(output, containsString("SMITHY_ROOT_DIR: " + Paths.get(".").toAbsolutePath().normalize()));
```

Comment:
Same a above, we can use instead `SmithyBuildUtils.getCurrentWorkingDirectory()`

Reply:
Yeah keeping it package-private for now